pkg/registry/core/resourceclaim/strategy_test.go

```diff
@@ -0,0 +1,76 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package resourceclaim
+
+import (
+	"testing"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	genericapirequest "k8s.io/apiserver/pkg/endpoints/request"
+	"k8s.io/kubernetes/pkg/apis/core"
+)
+
+func TestResourceClaimStrategy(t *testing.T) {
```

Comment:
from experience, testing all functions in a single test make it very hard to write scenarios.

At least split in two:
- one for PrepareForCreate and Validate
- one for PrepareForUpdate and ValidateUpdate

Same for the other strategies.

Reply:
I've split all three up into smaller, more logical tests.